books/RayTracingInOneWeekend.html

```diff
@@ -502,54 +502,103 @@
+Put another way, given an image with a 16&ratio;9 aspect ratio,
+
+  $$\text{width} / \text{height} = 16 / 9 = 1.7778$$
+
+As another example, an image 800 pixels wide by 400 pixels high has a 2&ratio;1 aspect ratio.
+
+The image's aspect ratio can be determined from the ratio of its height to its width.
+However, since we have a given aspect ratio in mind, it's easier to set the image's width and the
+  aspect ratio, and then using this to calculate for its height.
+This way, we can scale up or down the image by changing the image width, and it won't throw off our
+  desired aspect ratio.
+We do have to make sure that when we solve for the image height the resulting height is at least 1.
 
 In addition to setting up the pixel dimensions for the rendered image, we also need to set up a
-virtual viewport through which to pass our scene rays. For the standard pixel spacing (equally
-spaced in both the horizontal and vertical direction), the viewport's aspect ratio should be the
-same as our rendered image. We'll just pick a viewport two units in height. We'll also set the
-distance between the projection plane and the projection point to be one unit. This is referred to
-as the _focal length_, not to be confused with the _focus distance_, which we'll present later.
+  virtual _viewport_ through which to pass our scene rays.
+The viewport is a virtual rectangle in the 3D world that contains the grid of image pixel locations.
+If pixels are spaced the same distance horizontally as they are vertically, the viewport that
+  bounds them will have the same aspect ratio as the rendered image.
+The distance between two adjacent pixels is called the pixel spacing, and square pixels is the
+  standard.
+
+To start things off, we'll choose an arbitrary viewport height of 2.0, and scale the viewport width
+  to give us the desired aspect ratio.
+Here's a snippet of what this code will look like:
+
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ C++
+    auto aspect_ratio = 16.0 / 9.0;
+    int image_width = 400;
+
+    // Calculate the image height, and ensure that it's at least 1.
+    int image_height = static_cast<int>(image_width / aspect_ratio);
+    image_height = (image_height < 1) ? 1 : image_height;
+
+    // Calculate the viewport height. Since it's real valued, we don't check for values less then 1.
+    auto viewport_height = 2.0;
+    auto viewport_width = viewport_height * (static_cast<double>(image_width)/image_height);
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    [Listing [image-setup]: Rendered image setup]
+
+(If you're wondering why we don't just use `aspect_ratio` when computing `viewport_width`, it's
+because the _actual_ aspect ratio is the integer-based ratio of image width over image height, which
+will yield zero for values less than one.)
```

Comment:
Are you sure that's correct?

My understanding is that the reason we don't use the `aspect_ratio` is because it's just a user suggestion, and what really matters is the actual ratio of the image dimensions. I don't think that's quite what you're saying here?

Reply:
Yeah, it hurts my head when I try to parse this. Which doesn't really matter because it's completely bogus. Let me take another swipe.

Reply:
Changed to this:
> Note that `aspect_ratio` is an ideal ratio, which we approximate as best as possible with the
>   integer-based ratio of image width / image height.
> In order for our viewport proportions to exactly match our image proportions, we use the calculated
>   image aspect ratio to determine our final viewport width.

Reply:
Looks good